client-ts/Microsoft.AspNetCore.SignalR.Client.TS/HubConnection.ts

```diff
@@ -224,9 +224,18 @@ class Base64EncodedHubProtocol implements IHubProtocol {
     readonly type: ProtocolType;
 
     parseMessages(input: any): HubMessage[] {
+
+        // The format of the message is `size:message;`
+        let pos = input.indexOf(":");
+        if (pos == -1)
+        {
+            throw new Error("Invalid payload.");
+        }
+        let encodedMessage = input.substring(pos + 1, input.length - 1);
```

Comment:
It does now - it will fail for missing trailing `;` and if the message size does not match the length prefix.